src/org/ods/component/ScanWithAquaStage.groovy

```diff
@@ -0,0 +1,122 @@
+package org.ods.component
+
+import groovy.transform.TypeChecked
+import groovy.transform.TypeCheckingMode
+import org.ods.services.AquaService
+import org.ods.services.BitbucketService
+import org.ods.services.OpenShiftService
+import org.ods.util.ILogger
+
+@TypeChecked
+class ScanWithAquaStage extends Stage {
+
+    public final String STAGE_NAME = 'Aqua Security Scan'
+    private final AquaService aqua
+    private final BitbucketService bitbucket
+    private final OpenShiftService openShift
+    private final ScanWithAquaOptions options
+
+    @SuppressWarnings('ParameterCount')
+    @TypeChecked(TypeCheckingMode.SKIP)
+    ScanWithAquaStage(def script, IContext context, Map config, AquaService aqua, BitbucketService bitbucket,
+                      OpenShiftService openShift, ILogger logger) {
+        super(script, context, logger)
+        if (!config.resourceName) {
+            config.resourceName = context.componentId
+        }
+        this.options = new ScanWithAquaOptions(config)
+        this.aqua = aqua
+        this.bitbucket = bitbucket
+        this.openShift = openShift
+    }
+
+    protected run() {
+        Map connectionData = openShift.getConfigMapData(context.cdProject, "aqua")
+        // base URL of Aqua server.
+        String aquaUrl = connectionData['aqua.url']
+        if (!aquaUrl) {
+            steps.error "Please provide the URL of the Aqua platform!"
+        }
+        // name in Aqua of the registry that contains the image we want to scan
+        String registry = connectionData['aqua.registry']
+        if (!registry) {
+            steps.error "Please provide the name of the registry that contains the image of interest!"
+        }
+        // name of the credentials that stores the username/password of a user with access
+        // to the Aqua server identified by "aquaUrl", defaults to the cd-user
+        String secretName = connectionData['aqua.secretName']
+        String credentialsId = context.cdProject + "-"
+        if (!secretName) {
+            credentialsId = context.credentialsId
+            logger.info("No custom secretName was specified, continuing with default " +
```

Comment:
Maybe not entirely clear that this `secretName` can be set in the `ConfigMap`?

Reply:
added: ...was specified **in the ConfigMap**,...
the ConfigMap is explained now in the groovy doc